### PR TITLE
Switch from Java 8 to Java 11 in the README

### LINKED ISSUE #969: Switch requirements to Java 11
At least in the documentation, we can still test with Java 8 for now

### PR DESCRIPTION
**Pull request by DavideD**
We will continue to test with Java 8 for the time being
but at least we don't commit to it for the future.

And we are aligned to the base version used by the products.

Fixes #969

I will update the website before the release

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Learn more at <http://hibernate.org/reactive>.
 
 Hibernate Reactive has been tested with:
 
-- Java 8
+- Java 11
```
Comment (Sanne) [inline README.md:31]: since the wording is "has been tested with" , shouldn't we list Java 16 and 17 as well?

**Review by Sanne — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Sanne) [reply README.md:31]: @DavideD up

Comment (DavideD) [reply README.md]: Added 11,16 and 17

**Comment by DavideD** [comment]
Merged!

**Pull request merged**